internal/core/src/segcore/SegmentSealedImpl.cpp

```diff
@@ -142,13 +144,29 @@ SegmentSealedImpl::LoadScalarIndex(const LoadIndexInfo& info) {
 }
 
 void
-SegmentSealedImpl::LoadFieldData(const LoadFieldDataInfo& info) {
+SegmentSealedImpl::LoadFieldData(const LoadFieldDataInfo& load_info) {
     // print(info);
     // NOTE: lock only when data is ready to avoid starvation
-    AssertInfo(info.row_count > 0, "The row count of field data is 0");
-    auto field_id = FieldId(info.field_id);
-    AssertInfo(info.field_data != nullptr, "Field info blob is null");
-    auto size = info.row_count;
+    for (auto& field_info : load_info.field_infos) {  // only one field for now, parallel load field data in golang
+        auto info = field_info.second;
```

Comment:
use structured binding here to improve readability:
```
for (auto& [id, field_info] : load_info.field_infos)
```
also line 151 will copy the whole struct, which contains a vector of strings